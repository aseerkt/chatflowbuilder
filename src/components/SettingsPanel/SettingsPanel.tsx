import { TextNodeEditor } from '../TextNodeEditor'

type SettingsPanelProps = {
  nodeId: string
  nodeType: string
}

export default function SettingsPanel({
  nodeId,
  nodeType,
}: SettingsPanelProps) {
  switch (nodeType) {
    case 'textNode':
      return <TextNodeEditor nodeId={nodeId} />
    default:
      return null
  }
}
