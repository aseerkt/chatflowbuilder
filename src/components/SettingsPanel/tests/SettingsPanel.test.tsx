import { render } from '@testing-library/react'
import SettingsPanel from '../SettingsPanel'

describe('<SettingsPanel /> tests', () => {
  it('should match snapshot', () => {
    const { baseElement } = render(
      <SettingsPanel nodeId='1' nodeType='unknownType' />,
    )
    expect(baseElement).toMatchSnapshot()
  })
})
