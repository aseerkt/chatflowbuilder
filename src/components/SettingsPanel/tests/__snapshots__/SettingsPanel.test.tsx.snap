// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<SettingsPanel /> tests > should match snapshot 1`] = `
<body>
  <div />
</body>
`;
